src/nix/path-info.cc

```diff
@@ -87,41 +96,98 @@ struct CmdPathInfo : StorePathsCommand, MixJSON
             pathLen = std::max(pathLen, store->printStorePath(storePath).size());
 
         if (json) {
-            std::cout << store->pathInfoToJSON(
+            auto json = store->pathInfoToJSON(
                 // FIXME: preserve order?
                 StorePathSet(storePaths.begin(), storePaths.end()),
-                true, showClosureSize, SRI, AllowInvalid).dump();
+                true, showClosureSize, SRI, AllowInvalid);
+
+            if (showSubStatus) {
+                ThreadPool pool(fileTransferSettings.httpConnections);
+
+                Sync<nlohmann::json> jsonList_;
+
+                {
+                    auto jsonList(jsonList_.lock());
+                    *jsonList = nlohmann::json::array();
+                }
+
+                std::function<void(nlohmann::json::basic_json::value_type)> checkSubs;
+                checkSubs = [&](const nlohmann::json::basic_json::value_type & value) {
```

Comment:
same suggestion with the nested lambda expression here. all the type info is relatively noisy and can be reduced.